node/src/components/event_stream_server/tests.rs

```diff
@@ -0,0 +1,899 @@
+use std::{
+    cell::RefCell,
+    error::Error,
+    fs, io, iter, str,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use futures::{join, StreamExt};
+use http::StatusCode;
+use pretty_assertions::assert_eq;
+use rand::Rng;
+use schemars::schema_for;
+use tempfile::TempDir;
+use tokio::{
+    sync::{Barrier, Notify},
+    task::JoinHandle,
+    time,
+};
+use tracing::debug;
+
+use super::*;
+use crate::{logging, testing::TestRng};
+use sse_server::{
+    EventFilter, Id, QUERY_FIELD, SSE_API_MAIN_PATH, SSE_API_ROOT_PATH, SSE_API_SIGNATURES_PATH,
+};
+
+/// The total number of random events each `EventStreamServer` will emit by default, excluding the
+/// initial `ApiVersion` event.
+const EVENT_COUNT: u32 = 100;
+/// The event stream buffer length, set in the server's config.  Set to half of the total event
+/// count to allow for the buffer purging events in the test.
+const BUFFER_LENGTH: u32 = EVENT_COUNT / 2;
+/// The maximum amount of time to wait for a test server to complete.  If this time is exceeded, the
+/// test has probably hung, and should be deemed to have failed.
+const MAX_TEST_TIME: Duration = Duration::from_secs(1);
+/// The duration of the sleep called between each event being sent by the server.
+const DELAY_BETWEEN_EVENTS: Duration = Duration::from_millis(1);
+
+thread_local!(
+    /// A per-test identifier for the clients.
+    static CLIENT_ID: RefCell<usize> = RefCell::new(0)
```

Comment:
Is it possibly sufficient to use a random `u64` instead and not deal with any thread local stuff?

Reply:
Passing the RNG in here is painful, as in some tests this is called in parallel and the borrow checker has thoughts about that.  I'll pass a `&str` so we can keep with the current approach of having the clients named sequentially to make the log output in the tests more understandable.